hera_pspec/uvpspec.py

```diff
@@ -1090,6 +1165,13 @@ def check(self, just_meta=False):
                             except:
                                 raise AssertionError(err_msg)
 
+                    elif p in self._dicts_of_dicts:
+                        assert isinstance(getattr(self, p), (dict, odict))
+                        for k in getattr(self, p).keys():
+                            assert isinstance(getattr(self, p)[k], (dict, odict))
+                            for j in getattr(self, p)[k].keys():
+                                assert isinstance(getattr(self, p)[k][j], np.ndarray)
+                        
```

Comment:
should add here:
```
try:
    setattr(self, p, a.expected_type(getattr(self, p)))
except:
    raise AssertionError(err_msg)
```

Reply:
Ah yeah you are right, what I think I meant was something like
```
# dict of dicts
elif p in self._dicts_of_dicts:
    assert isinstance(getattr(self, p), (dict, odict))
    for k in getattr(self, p).keys():
        assert isinstance(getattr(self, p)[k], (dict, odict))
        for j in getattr(self, p)[k].keys():
            assert isinstance(getattr(self, p)[k][j], np.ndarray)
            try:
                setattr(self, getattr(self, p)[k][j], a.expected_type(getattr(self, p)[k][j]))
            except:
                raise AssertionError(err_msg)
```
which will work on a dictionary of dictionaries.